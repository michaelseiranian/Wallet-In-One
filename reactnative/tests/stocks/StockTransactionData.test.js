import React from 'react';
import { render, screen, waitFor} from '@testing-library/react-native';
import TransactionData from '../../screens/stocks/StockTransactionData';
import transaction from './fixtures/transaction.json';

/**
 * Tests for transaction data screen
 */
describe('<TransactionData />', () => {

    beforeEach(() => {
        global.fetch =  jest.fn( async (id) => {
            return Promise.resolve({
                status: 200, json: () => transaction
            })
        })
      })

      afterEach(() => {
        global.fetch.mockClear();
        delete global.fetch;
      })

    it('transactionData snapshot test', async () => {
        const transactionData = render(<TransactionData route={{params: "701"}} />);

        await waitFor( () => {
            const loading = screen.UNSAFE_queryByType('ActivityIndicator');
            expect(loading).toBeNull();
        })
        expect(transactionData).toMatchSnapshot();
    }
    )

    it('test transactionData fields', async () => {
        const transactionData = render(<TransactionData route={{params: "701"}} />);

        await waitFor( () => {
            const loading = screen.UNSAFE_queryByType('ActivityIndicator');
            expect(loading).toBeNull();
        })
        expect(screen.getByText("Name"))
        expect(screen.getByText("BUY NFLX DERIVATIVE"))
        expect(screen.getByText("Transaction ID"))
        expect(screen.getByText("WraX31pG9AuzLlKzAG8JIAAyNV9PZEuPaBq3k"))
        expect(screen.getByText("Amount"))
        expect(screen.getByText("£ 38.45"))
        expect(screen.getByText("Date"))
        expect(screen.getByText("2023-03-10"))
        expect(screen.getByText("Quantity"))
        expect(screen.getByText("4211.152345617756"))
        expect(screen.getByText("Price"))
        expect(screen.getByText("£ 0.01"))
        expect(screen.getByText("Fees"))
        expect(screen.getByText("£ 5"))
        expect(screen.UNSAFE_queryByType('Map')).toBeDefined()
    }
    )
});

describe('<TransactionData /> Incorrect ID', () => {
    beforeEach(() => {
        global.fetch =  jest.fn( async (id) => {
            return Promise.resolve({
                status: 400, json: () => null
            })
        })
      })

      afterEach(() => {
        global.fetch.mockClear();
        delete global.fetch;
      })

      it('test transactionData null', async () => {
        const transactionData = render(<TransactionData route={{params: "1"}} />);

        expect(screen.UNSAFE_queryByType('ActivityIndicator')).toBeDefined()
    }
    )
})
