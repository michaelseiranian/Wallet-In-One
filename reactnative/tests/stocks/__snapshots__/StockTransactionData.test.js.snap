// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransactionData /> transactionData snapshot test 1`] = `
<View
  style={
    [
      {
        "flex": 1,
        "padding": 10,
      },
      {
        "backgroundColor": "white",
        "flex": 1,
      },
    ]
  }
>
  <View
    style={
      {
        "flex": 1,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "black",
            "fontWeight": "700",
          },
          {
            "color": "black",
          },
        ]
      }
    >
      Name
    </Text>
    <Text
      style={
        {
          "color": "black",
        }
      }
    >
      BUY NFLX DERIVATIVE
      

    </Text>
    <Text
      style={
        [
          {
            "color": "black",
            "fontWeight": "700",
          },
          {
            "color": "black",
          },
        ]
      }
    >
      Transaction ID
    </Text>
    <Text
      style={
        {
          "color": "black",
        }
      }
    >
      WraX31pG9AuzLlKzAG8JIAAyNV9PZEuPaBq3k
      

    </Text>
    <Text
      style={
        [
          {
            "color": "black",
            "fontWeight": "700",
          },
          {
            "color": "black",
          },
        ]
      }
    >
      Amount
    </Text>
    <Text
      style={
        {
          "color": "black",
        }
      }
    >
      £ 
      38.45
      

    </Text>
    <Text
      style={
        [
          {
            "color": "black",
            "fontWeight": "700",
          },
          {
            "color": "black",
          },
        ]
      }
    >
      Date
    </Text>
    <Text
      style={
        {
          "color": "black",
        }
      }
    >
      2023-03-10
      

    </Text>
    <Text
      style={
        [
          {
            "color": "black",
            "fontWeight": "700",
          },
          {
            "color": "black",
          },
        ]
      }
    >
      Quantity
    </Text>
    <Text
      style={
        {
          "color": "black",
        }
      }
    >
      4211.152345617756
      

    </Text>
    <Text
      style={
        [
          {
            "color": "black",
            "fontWeight": "700",
          },
          {
            "color": "black",
          },
        ]
      }
    >
      Price
    </Text>
    <Text
      style={
        {
          "color": "black",
        }
      }
    >
      £ 
      0.01
      

    </Text>
    <Text
      style={
        [
          {
            "color": "black",
            "fontWeight": "700",
          },
          {
            "color": "black",
          },
        ]
      }
    >
      Fees
    </Text>
    <Text
      style={
        {
          "color": "black",
        }
      }
    >
      £ 
      5
      

    </Text>
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <View
        style={
          {
            "height": "100%",
          }
        }
      >
        <AIRMap
          initialRegion={null}
          onChange={[Function]}
          onMapReady={[Function]}
          region={
            {
              "latitude": 40.813,
              "latitudeDelta": 0.0922,
              "longitude": -109.962,
              "longitudeDelta": 0.0421,
            }
          }
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          testID="mapView"
        >
          <AIRMapMarker
            coordinate={
              {
                "latitude": 40.813,
                "longitude": -109.962,
              }
            }
            onPress={[Function]}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "position": "absolute",
                },
                undefined,
              ]
            }
            testID="marker"
          />
        </AIRMap>
      </View>
    </View>
  </View>
</View>
`;
